lasagne/layers/base.py

```diff
@@ -23,8 +23,10 @@ class Layer(object):
     network's output :class:`Layer` instance doubles as a handle to the full
     network.
     """
-    def __init__(self, input_layer):
-        self.input_layer = input_layer
+    def __init__(self, input_layer, name=None):
+        if input_layer is not None:
+            self.input_layer = input_layer
```

Comment:
`self.input_layer` should always exist as a property, shouldn't it? Also changing this will create a merge conflict in #52, so I'd appreciate leaving this line unchanged :)